src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -438,6 +495,10 @@ private static bool CheckConstraintsSingleType(TypeSymbol type, CheckConstraints
             Location location,
             DiagnosticBag diagnostics)
         {
+            // We do not report element locations in method parameters and return types
+            // so we will simply unwrap the type if it was a tuple.
```

Comment:
>// so we will simply unwrap the type if it was a tuple. [](start = 12, length = 55)

Please explain in the comment when and where do we check constraints for the "Rest" tuple, in the scenario, for which we hit this code path with a "long" tuple. I assume we are relying on TypeSymbolExtensions.VisitType to dig into the "Rest" tuple.